packages/prices/utils/price.ts

```diff
@@ -74,7 +76,11 @@ export const getCurrencyFromPriceResponse = (
 		| Record< string, never >
 		| CartShippingPackageShippingRate
 ): Currency => {
-	if ( ! currencyData || typeof currencyData !== 'object' ) {
+	if (
+		! currencyData ||
+		typeof currencyData !== 'object' ||
+		! currencyData?.currency_code
+	) {
```

Comment:
Would this be equivalent?
```suggestion
	if ( ! currencyData?.currency_code ) {
```